src/main/java/hlf/java/rest/client/controller/FabricClientController.java

```diff
@@ -55,6 +55,34 @@ public ResponseEntity<ClientResponseModel> postTransaction(
         networkName, contractName, transactionFunctionName, Optional.empty(), transactionParams);
   }
 
+  /**
+   * The REST Endpoint for writing a transaction to the ledger
+   *
+   * @param channelName String channel name
+   * @param chaincodeName String chaincode name
+   * @param peerNames String endorsing peers
+   * @param payload JSON String for arguments to the chaincode.
+   * @return responseEntity ResponseEntity Transaction Response
+   */
+  @PostMapping(value = "/init_transaction")
+  public ResponseEntity<ClientResponseModel> initTransaction(
+      @RequestHeader("channel") @Validated String channelName,
+      @RequestHeader("chaincode") @Validated String chaincodeName,
+      @RequestHeader("function") @Validated String functionName,
+      @RequestHeader(value = "peers", required = false) String peerNames,
+      @RequestBody @Validated String payload) {
+    log.info(
+        "Smart Contract init request Network Name: {}, Contract Name: {}, Function Name: {}, Endorsing Peers: {}, Transaction Parameters: {}",
+        functionName,
+        channelName,
+        chaincodeName,
+        peerNames,
+        payload);
+    List<String> lstPeerNames = Arrays.asList(peerNames.split(","));
```

Comment:
Null Pointer exeception could arise, as `peerNames` is not a required parameter

Reply:
Nice catch, added a condition.